packages/features/form-builder/FormBuilderField.tsx

```diff
@@ -0,0 +1,337 @@
+import { ErrorMessage } from "@hookform/error-message";
+import type { TFunction } from "next-i18next";
+import { Controller, useFormContext } from "react-hook-form";
+import type { z } from "zod";
+
+import { classNames } from "@calcom/lib";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { Label } from "@calcom/ui";
+import { Info } from "@calcom/ui/components/icon";
+
+import { Components, isValidValueProp } from "./Components";
+import { fieldTypesConfigMap } from "./fieldTypes";
+import type { fieldsSchema } from "./schema";
+
+type RhfForm = {
+  fields: z.infer<typeof fieldsSchema>;
+};
+
+type RhfFormFields = RhfForm["fields"];
+
+type RhfFormField = RhfFormFields[number];
+
+type ValueProps =
+  | {
+      value: string[];
+      setValue: (value: string[]) => void;
+    }
+  | {
+      value: string;
+      setValue: (value: string) => void;
+    }
+  | {
+      value: {
+        value: string;
+        optionValue: string;
+      };
+      setValue: (value: { value: string; optionValue: string }) => void;
+    }
+  | {
+      value: boolean;
+      setValue: (value: boolean) => void;
+    };
+
+export const FormBuilderField = ({
+  field,
+  readOnly,
+  className,
+}: {
+  field: RhfFormFields[number];
+  readOnly: boolean;
+  className: string;
+}) => {
+  const { t } = useLocale();
+  const { control, formState } = useFormContext();
+
+  const { hidden, placeholder, label } = getAndUpdateNormalizedValues(field, t);
```

Comment:
This is new logic which ensures that labels and placeholders, wherever they are, are set properly. If direct values are not set, default values from fieldTypeConfig is used.